app/src/main/java/com/svs/hztb/Activities/RegistrationActivity.java

```diff
@@ -22,31 +30,49 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_registration);
+        actionBarSettings();
+        initViews();
+    }
+
+    /**
+     * Initialize the views
+     */
+    private void initViews() {
+        mobileNumber = getView(R.id.editText_mobilePhoneNo);
+        String code = "+ ";
+        mobileNumber.setCompoundDrawablesWithIntrinsicBounds(new TextDrawable(code), null, null, null);
+        mobileNumber.setCompoundDrawablePadding(code.length() * 10);
+
+    }
+   /**
+      Action bar settings are updated
+    */
+    private void actionBarSettings() {
         getSupportActionBar().setDisplayOptions(ActionBar.DISPLAY_SHOW_CUSTOM);
         getSupportActionBar().setCustomView(R.layout.actionbar_title);
         setActionBarTitle(getString(R.string.title_activity_mobile_phone_registration));
-        mobileNumber = getView(R.id.editText_mobilePhoneNo);
     }
 
-    /*
-    @Override
-    public boolean dispatchKeyEvent(KeyEvent event) {
-        int keyaction = event.getAction();
-
-        if(keyaction == KeyEvent.ACTION_DOWN)
-        {
-            int keycode = event.getKeyCode();
-            int keyunicode = event.getUnicodeChar(event.getMetaState() );
-            char character = (char) keyunicode;
-            if (mobileNumber.getText().toString().length() == 0){
-                String number ="+"+mobileNumber.getText().toString();
-                Log.d(getPackageName().toString(),number);
-                mobileNumber.setText(number);
-            }
-
-        }
-
-        return super.dispatchKeyEvent(event);
+    /**
+    On click action performed
+     */
+   public void onSubmitClicked(View view){
+      if (isValidMobile()){
+            postDataForRegistration(mobileNumber.getText().toString());
+      }else {
+          displayMessage("Mobile Number Invalid");
+      }
+   }
+
+
+    /**
+     * Check if the mobile number entered is valid or not.
+     * @return
+     */
+    private boolean isValidMobile()
+    {
+        if (mobileNumber.getText().toString().length() > 9 && mobileNumber.getText().toString().length() <=12){
```

Comment:
Country Codes can be of 3 digits also. Hence, checking for 12 digits is not correct.